### PR TITLE
implementation of PrimitiveType using TDynamicType instead of Dynamic…

### PR DESCRIPTION
**Pull request by franava**
Changed the derivation of TPrimitiveTypes: now deriving from TDynamicType

### CODE REVIEW

File: include/dds/core/xtypes/UnionType.hpp [hunk 1]
```diff
@@ -48,18 +48,18 @@ class TUnionType : public TDynamicType<DELEGATE<T>>
 public:
     TUnionType(
         const std::string& name,
-        const PrimitiveType<T>& discriminator_type,
+        const TPrimitiveType<T, DELEGATE<T>>& discriminator_type,
```
Comment (lemunozm) [inline include/dds/core/xtypes/UnionType.hpp]: Are you sure about `Reference<T>`? (See the three occurrences)

**Comment by lemunozm** [comment]
Please attach the UML

File: include/dds/core/xtypes/UnionType.hpp (hunk 1 above)
Comment (franava) [reply include/dds/core/xtypes/UnionType.hpp]: that should be in line with current specs

File: include/dds/core/xtypes/PrimitiveTypes.hpp [hunk 2]
```diff
@@ -35,23 +35,24 @@ struct dynamic_type_traits
...
  * as follows:
  *
  * DynamicType int16Type = primitive_type<int16_t>();
  */
 template <typename T>
-PrimitiveType<T> primitive_type()
+TPrimitiveType<T, detail::DynamicType> primitive_type()
 {
-    static dds::core::xtypes::PrimitiveType<T> t;
+    static dds::core::xtypes::TPrimitiveType<T, detail::DynamicType> t;
```
Comment (lemunozm) [inline include/dds/core/xtypes/PrimitiveTypes.hpp]: Remove namespace

Comment (franava) [reply include/dds/core/xtypes/PrimitiveTypes.hpp]: done

File: include/dds/core/xtypes/PrimitiveTypes.hpp [hunk 3]
```diff
@@ -35,23 +35,24 @@ struct dynamic_type_traits
...
+
 /**
  * Primitive type constructor. This function can be used
  * as follows:
  *
  * DynamicType int16Type = primitive_type<int16_t>();
  */
 template <typename T>
-PrimitiveType<T> primitive_type()
+TPrimitiveType<T, detail::DynamicType> primitive_type()
```
Comment (lemunozm) [inline include/dds/core/xtypes/PrimitiveTypes.hpp]: I would create a PrimitiveType typename, and would set the primitive_type() function to return this type. What do you think?

Comment (lemunozm) [reply include/dds/core/xtypes/PrimitiveTypes.hpp]: Do not change by the moment

**Pull request merged**